node/lib/util/submodule_config_util.js

```diff
@@ -324,9 +325,25 @@ exports.initSubmoduleAndRepo = co.wrap(function *(repoUrl,
 
     const realUrl = exports.resolveSubmoduleUrl(repoUrl, url);
 
-    return yield NodeGit.Repository.initExt(subRepoDir, {
+    const result = yield NodeGit.Repository.initExt(subRepoDir, {
         originUrl: realUrl,
         workdirPath: workdirPath,
         flags: NO_DOTGIT_DIR | MKPATH | RELATIVE_GITLINK,
     });
+
+    // If there is a `submodule_template` directory in the git folder, copy its
+    // contents into the new `.git` directory for this submodule.
+    //
+    // TODO: fix NodeGit/libgit2 to support the `templatePath` argument.
+
+    const templatePath = path.join(repoPath, ".git", "submodule_template");
+    try {
+        yield fs.stat(templatePath);
+        yield (new Promise(callback => {
+            return ncp(templatePath, result.path(), callback);
+        }));
+    }
+    catch(e) {
```

Comment:
Is there a particular error you expect here?  Shouldn't we at least print a warning?